tests/phpunit/TemplateTest.php

```diff
@@ -1351,23 +1351,25 @@ public function testOpenAccessLookup7() : void {
   public function testSemanticScholar() : void {
    $text = "{{cite journal|doi=10.5555/555555}}";
    $template = $this->make_citation($text);
-   $return = $template->get_unpaywall_url($template->get2('doi'));
+   $return = $template->get_semanticscholar_url($template->get2('doi'));
    $this->assertSame('nothing', $return);
    $this->assertNull($template->get2('url'));
   }
   
+  // Test Unpaywall URL gets added
+  // DOI gets an URL on BHL 
   public function testUnPaywall1() : void {
-   $text = "{{cite journal}}";
+   $text = "{{cite journal|doi=10.1206/0003-0082(2006)3508[1:EEALSF]2.0.CO;2}}";
    $template = $this->make_citation($text);
-   $template->get_semanticscholar_url('10.1145/358589.358596', '');
-   $this->assertNull($template->get2('url'));
-   $this->assertSame('10.1145/358589.358596', $template->get2('doi'));
+   $template->get_unpaywall_url($template->get2('doi'), '');
+   $this->assertNotNull($template->get2('url'));
   }
-
+  
+  // Test Unpaywall OA URL does not get added when doi-access=free
   public function testUnPaywall2() : void {
-   $text = "{{cite journal|doi=10.1145/358589.358596|doi-access=free}}";
+   $text = "{{cite journal|doi=10.1007/s11075-018-0599-7}}";
```

Comment:
doi:10.1007/s11075-018-0599-7 is actually a case which should find an URL from http://arxiv.org/abs/1804.09042 , but I suppose that's added as parameter rather than as URL. I'm not actually sure it's that important to test that doi-access=free citations are left alone.